### ISSUE TITLE
When I use i18n and basePath at the same time, return 404.

### ISSUE THREAD

**Issue by whhjdi** [shuding/nextra#1111]
### What's the problem
When I use i18n and basePath at the same time, not working properly and return 404.
I don't know if this is a problem with next.js or a problem with nextra or  mine's problem.

<img width="600" alt="image" src="https://user-images.githubusercontent.com/22634897/209316779-a4f052ad-ca6c-4831-88f7-90412fac7133.png">

### How to reproduce

version : latest nextra-docs-template
This is the [demo repo](https://github.com/whhjdi/nextra-demo)

Any help? Thanks!

**Comment by B2o5T** [comment]
I can confirm this bug that goes from nextra middleware that doesn't handle `basePath`

**Comment by whhjdi** [comment]
I used the locales you wrote，it works very well, but when i switch language, the url does not include basePath at the beginning(xxx.com/index), and then rewrite to the path containing basePath（xxx.com/docs/index). 
May be when I click the button, it should include basePath.

**Comment by B2o5T** [comment]
Hm.. I tested it on swr-site example and it worked well, which version of Next.js do you use?

**Comment by whhjdi** [comment]
Yes, it works well on my computer, but in production env ,we have a gateway....
When I switched from English to Chinese(default locale)， the url is xxx.com/, and then rewrite to xxx.com/docs, Rewrite twice. 
and in production env xxx.com/ will not works 
this is the [repro repo](https://github.com/whhjdi/nextra-demo) 
Maybe should add basePath in language switch click event

**Comment by whhjdi** [comment]
The issue is closed. Can you help me reopen it? @B2o5T
